docs/extensibility/annotations.md

```diff
@@ -0,0 +1,69 @@
+# Annotations
+
+Annotations are a way to highlight a specific piece in a Gutenberg post. Examples of this include commenting on a piece of text and spellchecking. Both can use the annotations API to mark a piece of text.
+
+## API
+
+To see the API for yourself the easiest way is to have a block that is at least 200 characters long without formatting and putting the following in the console:
+
+```js
+wp.data.dispatch( 'core/editor' ).addAnnotation( {
+	source: "my-annotations-plugin",
```

Comment:
Should be a source? or an annotation name/id? I mean can a source have different types of annotations and how to distinguish between them?

Reply:
I left this fairly freeform so plugins can decide for themselves. If they want to group their block annotations with a different source than text annotations that is up to them.